binding/flutter/lib/porcupine.dart

```diff
@@ -0,0 +1,329 @@
+  /// calling `.sampleRate` to get the required sample rate and `.frameLength` to get the required frame size.
+  /// Audio must be single-channel and 16-bit linearly-encoded.
+  ///
+  /// Index of the detected keyword, or -1 if no detection occurred
+  int process(List<int> frame) {
+    if (_handle == null) {
+      throw new PvStateError(
+          "Attempted to process an audio frame after Porcupine was been deleted.");
+    }
+
+    if (frame == null) {
+      throw new PvArgumentError(
+          "Frame array provided to Porcupine process was null.");
+    }
+
+    if (frame.length != _frameLength) {
+      throw new PvArgumentError(
+          "Size of frame array provided to 'process' (${frame.length}) does not match the engine 'frameLength' ($_frameLength)");
+    }
+
+    // generate arguments for ffi
+    _cFrame.asTypedList(frame.length).setAll(0, frame);
+    Pointer<Int32> keywordIndexPtr = allocate(count: 1);
+
+    int status = _porcupineProcess(_handle, _cFrame, keywordIndexPtr);
+    if (status != 0) {
+      pvStatusToException(
+          status, "Porcupine failed to process an audio frame.");
+    }
+
+    return keywordIndexPtr.value;
+  }
+
+  /// Frees memory that was allocated for Porcupine
+  void delete() {
+    if (_handle != null) {
+      _porcupineDelete(_handle);
+      _handle = null;
+    }
+  }
+
+  static final _platform = Platform.isAndroid ? "android" : "ios";
+
+  static Future<void> _extractPorcupineResources() async {
+    _defaultModelPath =
+        await _extractResource("lib/common/porcupine_params.pv");
+
+    _builtInKeywordPaths = new Map();
+    String keywordAssetDir = "resources/keyword_files/$_platform";
+    for (var i = 0; i < BUILT_IN_KEYWORDS.length; i++) {
+      String keyword = BUILT_IN_KEYWORDS[i];
+      String keywordAssetPath = "$keywordAssetDir/${keyword}_$_platform.ppn";
+      String extractedPath = await _extractResource(keywordAssetPath);
+
+      if (extractedPath != null) {
+        _builtInKeywordPaths[keyword] = extractedPath;
+      }
+    }
+  }
+
+  static Future<String> _extractResource(String filePath) async {
+    String resourceDirectory = (await getApplicationDocumentsDirectory()).path;
+    String outputPath = '$resourceDirectory/$filePath';
+    File outputFile = new File(outputPath);
+
+    ByteData data = await rootBundle.load(filePath);
+    final buffer = data.buffer;
+
+    await outputFile.create(recursive: true);
+    await outputFile.writeAsBytes(
+        buffer.asUint8List(data.offsetInBytes, data.lengthInBytes));
+    return outputPath;
+  }
+}
+
+// loads lib
+final _porcupineLib = _load();
+DynamicLibrary _load() {
+  if (Platform.isAndroid) {
+    return DynamicLibrary.open("libpv_porcupine.so");
+  } else {
+    return DynamicLibrary.process();
+  }
+}
+
+// pv_porcupine_init
+typedef NativeInit = Int32 Function(Pointer<Utf8>, Int32,
+    Pointer<Pointer<Utf8>>, Pointer<Float>, Pointer<IntPtr>);
```

Comment:
Float -> Float32

Reply:
Dart FFI docs says that the Float type is 32-bit
![image](https://user-images.githubusercontent.com/2335311/101831079-56537380-3aea-11eb-9dd8-fc1122aa0d18.png)